Dockerfile

```
@@ -10,23 +12,27 @@ WORKDIR /app
 RUN groupadd --gid 1001 appuser && \
     useradd --create-home --uid 1001 --gid 1001 appuser
 
-# Copy requirements and install dependencies system-wide
-COPY src/requirements.txt .
+# Copy project metadata and install locked runtime dependencies
+COPY pyproject.toml uv.lock ./
 RUN apt-get update && \
```

Hadolint (2.14.0)

[warning] Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`

(DL3008)